### PR TITLE
Collect timezone and allow userAgentData to be overridden

### PR DESCRIPTION
**Pull request by danieljackins**
- [x] I've included a changeset (psst. run `yarn changeset`. Read about changesets [here](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md)).

### CODE REVIEW

**Comment by silesky** [comment]
@danieljackins CI is failing FYI

File: packages/browser/src/plugins/page-enrichment/index.ts [hunk 1]
```diff
@@ -214,6 +215,14 @@ class PageEnrichmentPlugin implements Plugin {
 
     evtCtx.userAgent = navigator.userAgent
 
+    try {
+      evtCtx.userAgentData = await clientHints(
+        this.instance.options.highEntropyValuesClientHints
```
Comment (chrisradek) [inline packages/browser/src/plugins/page-enrichment/index.ts:220]: Similar to the original PR, can we cache the client hints during the plugin load? These fields won't change, so no reason to perform a potentially expensive operation on every event :smile:

File: packages/browser/src/plugins/page-enrichment/__tests__/index.test.ts [hunk 2]
```diff
@@ -283,6 +288,11 @@ describe('Other visitor metadata', () => {
     }
   })
 
+  it('should add .timezone', async () => {
+    const ctx = await analytics.track('test')
+    assert(ctx.event.context?.timezone)
```
Comment (silesky) [inline packages/browser/src/plugins/page-enrichment/__tests__/index.test.ts:293]: can we assert that this is a string / actual TZ?

**Review by silesky — APPROVED** [review]
LGTM

**Review by chrisradek — APPROVED** [review]
:shipit:

**Pull request merged**